keras_contrib/applications/wide_resnet.py

```diff
@@ -0,0 +1,300 @@
+
+    if dropout > 0.0:
+        x = Dropout(dropout)(x)
+
+    x = Convolution2D(16 * k, 3, 3, border_mode='same')(x)
+    x = BatchNormalization(axis=channel_axis)(x)
+    x = Activation('relu')(x)
+
+    m = merge([init, x], mode='sum')
+    return m
+
+
+def __conv2_block(input, k=1, dropout=0.0):
+    init = input
+
+    channel_axis = 1 if K.image_dim_ordering() == "th" else -1
+
+    # Check if input number of filters is same as 32 * k, else create convolution2d for this input
+    if K.image_dim_ordering() == "th":
+        if init._keras_shape[1] != 32 * k:
+            init = Convolution2D(32 * k, 1, 1, activation='linear', border_mode='same')(init)
+    else:
+        if init._keras_shape[-1] != 32 * k:
+            init = Convolution2D(32 * k, 1, 1, activation='linear', border_mode='same')(init)
+
+    x = Convolution2D(32 * k, 3, 3, border_mode='same')(input)
+    x = BatchNormalization(axis=channel_axis)(x)
+    x = Activation('relu')(x)
+
+    if dropout > 0.0:
+        x = Dropout(dropout)(x)
+
+    x = Convolution2D(32 * k, 3, 3, border_mode='same')(x)
+    x = BatchNormalization(axis=channel_axis)(x)
+    x = Activation('relu')(x)
+
+    m = merge([init, x], mode='sum')
+    return m
+
+
+def __conv3_block(input, k=1, dropout=0.0):
+    init = input
+
+    channel_axis = 1 if K.image_dim_ordering() == "th" else -1
+
+    # Check if input number of filters is same as 64 * k, else create convolution2d for this input
+    if K.image_dim_ordering() == "th":
+        if init._keras_shape[1] != 64 * k:
+            init = Convolution2D(64 * k, 1, 1, activation='linear', border_mode='same')(init)
+    else:
+        if init._keras_shape[-1] != 64 * k:
+            init = Convolution2D(64 * k, 1, 1, activation='linear', border_mode='same')(init)
+
+    x = Convolution2D(64 * k, 3, 3, border_mode='same')(input)
+    x = BatchNormalization(axis=channel_axis)(x)
+    x = Activation('relu')(x)
+
+    if dropout > 0.0:
+        x = Dropout(dropout)(x)
+
+    x = Convolution2D(64 * k, 3, 3, border_mode='same')(x)
+    x = BatchNormalization(axis=channel_axis)(x)
+    x = Activation('relu')(x)
+
+    m = merge([init, x], mode='sum')
+    return m
+
+
+def __create_wide_residual_network(nb_classes, img_input, include_top, depth=28, width=8, dropout=0.0):
+    ''' Creates a Wide Residual Network with specified parameters
+
+    Args:
+        nb_classes: Number of output classes
+        img_input: Input tensor or layer
+        include_top: Flag to include the last dense layer
+        depth: Depth of the network. Compute N = (n - 4) / 6.
+               For a depth of 16, n = 16, N = (16 - 4) / 6 = 2
+               For a depth of 28, n = 28, N = (28 - 4) / 6 = 4
+               For a depth of 40, n = 40, N = (40 - 4) / 6 = 6
+        width: Width of the network.
+        dropout: Adds dropout if value is greater than 0.0
+
+    Returns:a Keras Model
+
+    '''
+
+    N = (depth - 4) // 6
+
+    x = __initial_conv(img_input)
```

Comment:
Better to change '__initial_conv', '__conv1_block', '__conv2_block', __conv3_block to '__conv1_block', __conv2_block, '__conv3_block', '__conv4_block' respectively to keep the same as in the paper.